src/computations/forces/ForceComputations.cpp

```diff
@@ -47,11 +47,23 @@ void ForceComputations::computeLennardJonesPotential(ParticleContainer &particle
 }
 
 void ForceComputations::resetForces(ParticleContainer &particles) {
-    for (Particle &particle: particles) {
+    for (auto it = particles.begin(); *it != *(particles.end()); it->operator++()) {
+        Particle& particle = **it;
         particle.setOldF(particle.getF());
         particle.setF({0, 0, 0});
```

Comment:
I haven't tested it, but wouldn't something like this without pointer not work? I would avoid using pointers, and definitely not raw pointers in modern C++. Use [smart pointers](https://www.geeksforgeeks.org/smart-pointers-cpp/) if really necessary.
```suggestion
    for (auto it = particles.begin(); it != particles.end(); it->operator++()) {
        it->setOldF(particle.getF());
        it->setF({0, 0, 0});
```

Reply:
This is what it originally looked like. The problem came when we introduced polymorphism into the ParticleContainer. The only way to make it so that the begin function can return an iterator for the direct sum method, as well as for the linked cell method was to use a unique pointer to an abstract ParticleIterator superclass.
That's why there is an extra dereference on the it variable.
This was my first time dealing with polymorphism in cpp so i may be overlooking something. But for me it this was the only way to make it work. If you have any insights, that'd be much appreciated